### PR TITLE
College entity implementation.

### PR DESCRIPTION
**Pull request by sumitdhungana14**

### CODE REVIEW

File: src/main/java/api/college/controller/CollegeController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Get("/")
+    public Iterable<College> index() {
+        return collegeServices.showAllCollege();
+    }
+
+    @Post("/addCollege")
+    public HttpResponse addCollege(@Body @Valid College college){
+        collegeServices.add(college);
+        return HttpResponse.ok();
```
Comment (vsujan) [inline src/main/java/api/college/controller/CollegeController.java:31]: After the college entity is added successfully to the database, return httpresponse with the college information in the Http body.

File: src/main/java/api/college/models/College.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity(name = "college")
+public class College {
+
+    @Id
+    @GeneratedValue
```
Comment (vsujan) [inline src/main/java/api/college/models/College.java:13]: strategy is missing here.

File: src/main/java/api/user/services/UserService.java [hunk 3]
```diff
@@ -22,4 +20,4 @@
     void deleteById(int id);
 
     User editUser(User user);
-}
+}
```
Comment (vsujan) [inline src/main/java/api/user/services/UserService.java]: Extra line here.

File: src/main/java/api/user/models/User.java [hunk 4]
```diff
@@ -39,4 +49,16 @@ public void setEmail(String email){
...
+    }
+
+    public void setCollege(College college) {
+        this.college = college;
+    }
+
+    public int getCollegeId() {
+        return collegeId;
+    }
 }
```
Comment (vsujan) [inline src/main/java/api/user/models/User.java]: Extra line here.

File: src/main/java/api/user/controller/UserController.java [hunk 5]
```diff
@@ -29,6 +34,11 @@ public HttpResponse getUserbyId(int id){
 
     @Post("/addUser")
     public HttpResponse addUser(@Body @Valid User user){
+        //fetch college by collegeId
```
Comment (vsujan) [inline src/main/java/api/user/controller/UserController.java:37]: Can we move this logic to service?

File: src/main/java/api/user/controller/UserController.java [hunk 6]
```diff
@@ -1,22 +1,27 @@
...
+import java.util.Optional;
 
 @Controller("/user")
 public class UserController {
 
     @Inject
     UserService userService;
 
+    @Inject
+    CollegeServices collegeServices;
```
Comment (vsujan) [inline src/main/java/api/user/controller/UserController.java:23]: Typo CollegeServices, collegeServices.

File: src/main/java/api/college/services/CollegeServices.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    Optional<College> findById(int id);
+
+    Iterable<College> showAllCollege();
+
+    void add(College college);
+
+    void deleteById(int id);
+
+    College editCollege(College college);
+}
```
Comment (vsujan) [inline src/main/java/api/college/services/CollegeServices.java]: New line here.

File: src/main/java/api/user/controller/UserController.java (hunk 5 above)
Comment (sumitdhungana14) [reply src/main/java/api/user/controller/UserController.java:37]: Resolved.

**Pull request merged**